lib/src/kbasesearchengine/parse/ObjectParser.java

```diff
@@ -57,21 +72,47 @@
         return guidToJson;
     }
 
-    public static GUID prepareGUID(ObjectTypeParsingRules parsingRules,
-            GUID guid, ObjectJsonPath path,
+    /** Appends the sub-object path and id to the specified GUID if the parsing
+     * rules define the extraction of data from a sub-object.
+     *
+     *
+     * @param parsingRules parsing rules
+     * @param guid GUID without the sub-object path and id
+     * @param path
+     * @param idConsumer
+     * @return GUID with sub-object path and id if data needs to be extracted
+     * from sub-object, lese GUID with just main object information.
+     */
+    public static GUID prepareGUID(
+            ObjectTypeParsingRules parsingRules,
+            GUID guid,
+            ObjectJsonPath path,
             SimpleIdConsumer idConsumer) {
-        String innerSubType = null;
-        String innerID = null;
+        String subObjectType = null;
+        String subObjectID = null;
         if (parsingRules.getSubObjectPath().isPresent()) {
-            innerID = idConsumer.getPrimaryKey() == null ? path.toString() : 
+            subObjectID = idConsumer.getPrimaryKey() == null ? path.toString() :
                 String.valueOf(idConsumer.getPrimaryKey());
-            innerSubType = parsingRules.getSubObjectType().get();
+            subObjectType = parsingRules.getSubObjectType().get();
         }
-        return new GUID(guid, innerSubType, innerID);
+        return new GUID(guid, subObjectType, subObjectID);
     }
-    
-    public static String extractParentFragment(ObjectTypeParsingRules parsingRules,
-            JsonParser jts) throws ObjectParseException, IOException {
+
+    /** Returns null if the specified parsing rules do not have rules for
+     * extracting parent elements. Else returns "/" for the root path of the
+     * parent json object.
```

Comment:
I don't think this is true - I believe what's returned is a JSON string containing the extracted portions of the parent object.

Reply:
You are right! I overlooked the nested calls to pathToJson. Documentation has been fixed.